src/services/organizationService.js

```diff
@@ -34,45 +34,42 @@ async function registerOrganization(organizationData) {
 async function loginOrganization(email, password) {
   const organization = await Organization.findOne({ email });
   if (!organization) {
-    throw new Error("Invalid email or password!!!");
+    throw new Error('Invalid email or password!');
   }
   if (organization.isDeleted == true) {
-    throw new Error("This account has been deleted, please contact support");
+    throw new Error('This account has been deleted, please contact support');
   }
 
   const match = await bcrypt.compare(password, organization.hashedPassword);
 
   if (!match) {
-    throw new Error("Invalid email or password!!!");
+    throw new Error('Invalid email or password!');
   }
   return createToken(organization);
 }
 
-async function updateOrganization(id, requestBody) {
+async function updateOrganization(id, requestBody, isAdmin) {
   const existingOrganization = await Organization.findById(id);
-  existingOrganization.name = requestBody.name
-    ? requestBody.name
-    : existingOrganization.name;
-  existingOrganization.email = requestBody.email
-    ? requestBody.email
-    : existingOrganization.email;
-  existingOrganization.managerFirstName = requestBody.managerFirstName
-    ? requestBody.managerFirstName
-    : existingOrganization.managerFirstName;
-  existingOrganization.managerLastName = requestBody.managerLastName
-    ? requestBody.managerLastName
-    : existingOrganization.managerLastName;
-  existingOrganization.phone = requestBody.phone
-    ? requestBody.phone
-    : existingOrganization.phone;
-  existingOrganization.region = requestBody.region
-    ? requestBody.region
-    : existingOrganization.region;
-  existingOrganization.address = requestBody.address
-    ? requestBody.address
-    : existingOrganization.address;
+  if (!existingOrganization) {
+    throw new Error('Organization not found');
+  }
+
+  for (let key of Object.keys(requestBody)) {
+    if (key == 'email' && requestBody[key] == '') {
```

Comment:
what if key == 'isDeleted' from hacked front end and !isAdmin ?

Reply:
isAdmin cames after parsing token
isDeleted will be added